src/Umbraco.Web/Routing/UrlProvider.cs

```diff
@@ -210,17 +208,20 @@ public string GetUrl(int id, UrlProviderMode mode)
         /// </remarks>
         public string GetUrl(int id, Uri current, UrlProviderMode mode)
         {
-            var url = _urlProviders.Select(provider => provider.GetUrl(_umbracoContext, id, current, mode))
+            string url = _urlProviders.Select(provider => provider.GetUrl(_umbracoContext, id, current, mode))
                 .FirstOrDefault(u => u != null);
             return url ?? "#"; // legacy wants this
         }
 
         internal string GetUrlFromRoute(int id, string route)
         {
-            var provider = _urlProviders.OfType<DefaultUrlProvider>().FirstOrDefault();
-            var url = provider == null 
-                ? route // what else?
-                : provider.GetUrlFromRoute(route, UmbracoContext.Current, id, _umbracoContext.CleanedUmbracoUrl, Mode);
+            IUrlProvider provider = _urlProviders.OfType<IUrlProvider>().FirstOrDefault();
+
+            var url = provider is DefaultUrlProvider

```

Comment:
Instead of checking with `is` and then casting, it's more performant to use:
```c#
var defaultProvider = provider as DefaultUrlProvider;
if (defaultProvider != null) {}
else {}
```
Or when using C# 7 [pattern matching](https://docs.microsoft.com/en-us/dotnet/csharp/pattern-matching):
```c#
if (provider is DefaultUrlProvider defaultProvider) {}
else {}
```